src/Orleans.Core.Abstractions/Core/Internal/IGrainManagementExtension.cs

```diff
@@ -12,6 +12,12 @@ public interface IGrainManagementExtension : IGrainExtension
         /// Deactivates the current instance once it becomes idle.
         /// </summary>
         /// <returns>A <see cref="Task"/> which represents the method call.</returns>
-        Task DeactivateOnIdle();
+        ValueTask DeactivateOnIdle();
+
+        /// <summary>
+        /// Attempts to migrate the current instance to a new location once it becomes idle.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> which represents the method call.</returns>
+        ValueTask MigrateOnIdle();
```

Comment:
I wonder if we should call it `TryMigrateOnIdle` since it's not guaranteed, if people don't read the doc :p

Reply:
Good question. It not returning a `bool` is my only issue, but I suppose we could make it async and make it return a `Task<bool>` or `Task<SiloAddress>` (prefer not this). Then again, would those be actionable results?

Reply:
Na it's fine for now imo